packages/appeal-reply-service-api/Dockerfile

```diff
@@ -1,30 +1,38 @@
-FROM node:14
-ARG APP_NAME
-ARG BUILD_ID
-ARG VERSION
-WORKDIR /opt/app
-ENV APP_NAME="${APP_NAME}"
-ENV BUILD_ID="${BUILD_ID}"
-ENV VERSION="${VERSION}"
-RUN mkdir -p /opt/app/data
-
-# This must be built locally from the common package
-COPY --from=pinscommonukscontainers3887default.azurecr.io/common:latest /opt/app ../common
-COPY --from=pinscommonukscontainers3887default.azurecr.io/business-rules:latest /opt/app ../business-rules
-ADD api api
-ADD src src
-ADD __tests__ __tests__
-ADD jest.config.js jest.config.js
-
-# ADD node_modules node_modules
-ADD package.json package.json
-ADD package-lock.json package-lock.json
+# Stage 1/3: Build
+FROM node:14-alpine AS build
+
+WORKDIR /opt/app 
+
+COPY packages/common ../common
+COPY packages/common ../business-rules
+COPY packages/appeal-reply-service-api/package*.json ./
+COPY packages/appeal-reply-service-api/api ./api
+COPY packages/appeal-reply-service-api/src ./src
+
 RUN npm i --production
-RUN npm prune --production \
-  && npm rebuild \
-  && npm version ${VERSION} --no-git-tag-version --allow-same-version || true
+
+# Stage 2/3: Run Tests
+FROM node:14 AS test
+
+WORKDIR /opt/app 
+
+COPY --from=build /opt/app ./
+
+COPY packages/appeals-service-api/__tests__ ./__tests__
+COPY packages/appeals-service-api/jest.config.js ./jest.config.js
+COPY packages/appeals-service-api/jest-mongodb-config.js ./jest-mongodb-config.js
+
 RUN npm run documentation; echo $? > /npm.exitcode
+
+# Stage 3/3: App Run
+FROM node:14-alpine AS app
```

Comment:
If we are changing the source image here we should change in docker-compose too to keep the sane

Reply:
This one is actually Alpine in docker-compose! But you're right, I'll check the rest.